### DISCUSSION TITLE
the unit of socket_timeout and socket_connect_timeout?

### DISCUSSION THREAD

**Discussion by zhaoyi2** [Grokzen/redis-py-cluster#429] [Q&A] [answered]
who knows the unit of socket_timeout and socket_connect_timeout in RedisCluster, seconds or milliseconds?

**Comment by Grokzen** [comment, marked as answer]
@zhaoyi2 Thanks for being the first person to use the Discussions page for this.

`socket_timeout` is an option that is simply passed along upstream to redis-py here via the `kwargs` parameter https://github.com/Grokzen/redis-py-cluster/blob/master/rediscluster/client.py#L305

It then moves into the connection pool here https://github.com/andymccurdy/redis-py/blob/1a41cfd95a53a95b078084d8627be6b6fba3bb71/redis/client.py#L739

And the value will finally end up here https://github.com/andymccurdy/redis-py/blob/e635130043d7657d9bee6911995f649e2de9eb04/redis/connection.py#L596 inside redis-py and be a value that is set on the regular python socket object.

Looking into the documentation for sockets inside python you find this here for python 3.9 (Change version to match your version you use)

https://docs.python.org/3/library/socket.html#socket.socket.settimeout

```
socket.settimeout(value)
Set a timeout on blocking socket operations. The value argument can be a nonnegative floating point number expressing seconds, or None. If a non-zero value is given, subsequent socket operations will raise a timeout exception if the timeout period value has elapsed before the operation has completed. If zero is given, the socket is put in non-blocking mode. If None is given, the socket is put in blocking mode.

For further information, please consult the notes on socket timeouts.

Changed in version 3.7: The method no longer toggles SOCK_NONBLOCK flag on socket.type.
```

Sot he answer to your question is `... expresses in seconds or None`

**Reply by zhaoyi2** [reply]
Thank you for your reply. It really helped me a lot.
